ui/components/Navigator.js

```diff
@@ -703,70 +708,73 @@ class Navigator extends React.Component {
 
     return content;
-  }
-  updatenavigatorComponentsMenus() {
-    const self = this;
-    const { navigatorComponents } = this.state;
-    navigatorComponents.forEach((cat, ind) => {
-      if (cat.id === LIFECYCLE) {
-        cat.children.forEach((catc, ind1) => {
-          if (catc.id == SERVICE_MESH) {
-            return;
-          }
-          const icon = self.pickIcon(catc.id, catc.href);
-          navigatorComponents[ind].children[ind1].icon = icon;
+  };
 
-          const cr = self.fetchChildren(catc.id);
-          navigatorComponents[ind].children[ind1].children = cr;
-        });
-      }
+  const updatenavigatorComponentsMenus = () => {
+    setNavigatorComponents((prevState) => {
+      prevState.forEach((cat, ind) => {
+        if (cat.id === LIFECYCLE) {
+          cat.children.forEach((catc, ind1) => {
+            if (catc.id == SERVICE_MESH) {
+              return;
+            }
+            const icon = pickIcon(catc.id, catc.href);
+            prevState[ind].children[ind1].icon = icon;
 
-      if (cat.id === 'Configuration') {
-        let show = false;
-        cat.children?.forEach((ch) => {
-          if (ch.id === 'Designs') {
-            const idx = self.props.capabilitiesRegistry?.capabilities?.findIndex(
-              (cap) => cap.feature === 'persist-meshery-patterns',
-            );
-            if (idx != -1) {
-              ch.show = true;
-              show = true;
+            const cr = fetchChildren(catc.id);
+            prevState[ind].children[ind1].children = cr;
+          });
+        }
+
+        if (cat.id === 'Configuration') {
+          let show = false;
+          cat.children?.forEach((ch) => {
+            if (ch.id === 'Designs') {
+              const idx = props.capabilitiesRegistry?.capabilities?.findIndex(
+                (cap) => cap.feature === 'persist-meshery-patterns',
+              );
+              if (idx != -1) {
+                ch.show = true;
+                show = true;
+              }
             }
-          }
-        });
+          });
 
-        cat.show = show;
-      }
+          cat.show = show;
+        }
+      });
+      return prevState;
     });
-  }
-  updateAdaptersLink() {
-    const { navigatorComponents } = this.state;
-    navigatorComponents.forEach((cat, ind) => {
-      if (cat.id === LIFECYCLE) {
-        cat.children.forEach((catc, ind1) => {
-          if (
-            typeof navigatorComponents[ind].children[ind1].children[0] !== 'undefined' &&
-            typeof navigatorComponents[ind].children[ind1].children[0].href !== 'undefined'
-          ) {
-            const val = true;
-            const newhref = `${navigatorComponents[ind].children[ind1].children[0].href}`;
-            navigatorComponents[ind].children[ind1].link = val;
-            navigatorComponents[ind].children[ind1].href = newhref;
-          }
-        });
-      }
+  };
+
+  const updateAdaptersLink = () => {
+    setNavigatorComponents((prevState) => {
```

Comment:
Are we seeting that navigatorComponents here because I didn't see we are doing it in above code that you removed? can you share why or I'm missing something

Reply:
in upadateAdaptersLink, navigatorComponents state is being changed through forEach loop(in removed code of class component). So I tried to update the state using setNavigatorComponents .

Reply:
I have reverted the changes made here. I have removed the setState.